packages/devcmd/src/utils/type_utils.ts

```diff
@@ -0,0 +1,3 @@
+export function isString(x: any): x is string {
+  return typeof x === "string";
```

Comment:
I'm still not sure why we would want such trivial type guards.

I can understand why typeguards that kind of mimic types like [Result in Rust](https://doc.rust-lang.org/std/result/) are useful but writing ```if(isString(x))``` doesnt really seem better than just typing out ````if(typeof x ==='string')``` to me - but correct me if I'm wrong 😄

Reply:
it's mostly a question of taste for many types (I personally find `isString(x)` considerably more readable than `typeof x === "string"`).

For the special case of `undefined`, there's a common mistake of writing `typeof x === undefined` (i.e. forgetting the quotes around "undefined"), so I'm strongly for an easier-to-use type guard in this case.
As an extension, I usually also use consistent type guards for other types like string or number.